core/cache/changes.go

```diff
@@ -48,3 +48,30 @@ type RemoveApplication struct {
 	ModelUUID string
 	Name      string
 }
+
+// UnitChange represents either a new unit, or a change
+// to an existing unit in a model.
+type UnitChange struct {
+	ModelUUID      string
+	Name           string
+	Application    string
+	Series         string
+	CharmURL       string
+	PublicAddress  string
+	PrivateAddress string
+	MachineId      string
+	// TODO (manadart 2019-02-11): How should we handle these given that
```

Comment:
@howbazaar What should we do here? 

I suppose there is a case for moving the network package under core insofar as it does not have other transient juju dependencies. Perhaps gradually?